app/scripts/modules/core/src/pipeline/config/triggers/artifacts/expectedArtifact.component.ts

```diff
@@ -0,0 +1,50 @@
+import { IComponentController, IComponentOptions, module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IExpectedArtifact } from 'core/domain/IExpectedArtifact';
+import { IPipeline } from 'core/domain/IPipeline';
+import { IArtifact } from 'core/domain/IArtifact';
+
+class ExpectedArtifactController implements IComponentController {
+  public expectedArtifact: IExpectedArtifact;
+  public pipeline: IPipeline;
+
+  public removeArtifact(): void {
+    const artifactIndex = this.pipeline.expectedArtifacts.indexOf(this.expectedArtifact);
+    this.pipeline.expectedArtifacts.splice(artifactIndex, 1);
```

Comment:
Feels odd to do a splice here and a filter below, but not a big deal.

Reply:
Will filter here too - was taken from prior code